fe/fe-core/src/main/java/com/starrocks/datacache/DataCacheSelectExecutor.java

```diff
@@ -0,0 +1,86 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.datacache;
+
+import com.google.common.base.Preconditions;
+import com.starrocks.common.UserException;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.qe.SessionVariable;
+import com.starrocks.qe.StmtExecutor;
+import com.starrocks.qe.scheduler.Coordinator;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.sql.ast.DataCacheSelectStatement;
+import com.starrocks.sql.ast.InsertStmt;
+import com.starrocks.system.Backend;
+import com.starrocks.system.SystemInfoService;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class DataCacheSelectExecutor {
+    private static final Logger LOG = LogManager.getLogger(DataCacheSelectExecutor.class);
+
+    public static Optional<DataCacheSelectMetrics> cacheSelect(DataCacheSelectStatement statement,
+                                                             ConnectContext connectContext) throws Exception {
+        // backup original session variable
+        SessionVariable sessionVariableBackup = connectContext.getSessionVariable();
+        // clone an new session variable
+        SessionVariable tmpSessionVariable = (SessionVariable) connectContext.getSessionVariable().clone();
+        // force enable datacache and populate
+        tmpSessionVariable.setEnableScanDataCache(true);
+        tmpSessionVariable.setEnablePopulateDataCache(true);
+        connectContext.setSessionVariable(tmpSessionVariable);
```

Comment:
Also need set `async_populate_mode` to false, for warmup query, it is expected to try cache all read data in synchronous mode.

Reply:
done